### DISCUSSION TITLE
Implementation of non-free weather sources

### DISCUSSION THREAD

**Discussion by neunzehnachtneun** [breezy-weather/breezy-weather#626] [Q&A] [answered]
### Describe your idea

Currently, non-free APIS contradict the contribution guidelines. I am a meteorologist and like to use the data from kachelmannwetter. There is an API with access to comprehensive station and forecast data, but unfortunately this is only available to paid accounts.

Hence two questions:
First, is the implementation of a non-free API possible? I have read this as an exclusion argument, but I would be happy if it were still possible. I would also support the integration, but may need some help.
Secondly, if this is not possible, what do you think of the idea of allowing manual integration of additional APIs within the app?

Thanks for the great work so far!

Link to kachelmannwetter-API: https://api.kachelmannwetter.com/v02/_doc.html

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] My idea is not related to a question covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document, or a feature described in the [Homepage](https://github.com/breezy-weather/breezy-weather/blob/main/docs/HOMEPAGE.md) document.
- [X] I have updated the app to **[version 4.6.14 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment, marked as answer]
Integration of paid API is totally possible. It's just that it won't make it to Breezy Weather main repo, because there is no way for us to be able to maintain the API as we don't want to pay, besides the fact that it's time-consuming per number of users using it.

So, you can make your own source and maintain it on your own fork. I know at least two other users doing that. It's easy to rebase regularly with latest changes, as you will create your own files for your source in a separate folder in the code.

**Comment by neunzehnachtneun** [comment]
Understand, thanks a lot!

Where do I find additional information to build the app?

**Reply by papjul** [reply]
Sorry I missed your message.

Here's the contribution guide: https://github.com/breezy-weather/breezy-weather/blob/main/CONTRIBUTE.md#weather-sources

There are a few more features that were added since it was written that you can find by looking into other sources, but that should get you started
